compiler/include/expr.h

```diff
@@ -40,6 +40,8 @@ class Expr : public BaseAST {
   virtual bool    isStmt()                                           const;
   virtual QualifiedType qualType();
   virtual void    verify();
+          void    verify(AstTag expectedTag);
+          void    verifyParent(const Expr* child, const char* field);
```

Comment:
Could you comment these methods here to indicate their purpose?
In particular I can see what the AstTag version is doing from the source code, but I don't understand the purpose of `field` in verifyParent. Given that it will raise an INT_FATAL, why does it need a customizeable error message at all?